### PR TITLE
Feature/quality pipeline GitHub

### PR DESCRIPTION
**Pull request by ultymatom**
Adding the quality workflow for github

### CODE REVIEW

**Comment by ultymatom** [comment]
Draft for two reason:

- Waiting for github pipeline generator PR to be merged
- waiting for the quality for devon4node PR to be merged also

File: documentation/src/azure-devops/setup-quality-pipeline.asciidoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+:provider: Azure Devops
```
Comment (albertdb) [inline documentation/src/azure-devops/setup-quality-pipeline.asciidoc:1]:
```suggestion
:provider: Azure DevOps

```

File: documentation/src/github/setup-quality-pipeline.asciidoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+:provider: Github
```
Comment (albertdb) [inline documentation/src/github/setup-quality-pipeline.asciidoc:1]:
```suggestion
:provider: GitHub

```

File: documentation/src/github/setup-quality-pipeline.asciidoc [hunk 3]
```diff
@@ -0,0 +1,10 @@
+:provider: Github
+:pipeline_type: workflow
+:trigger_sentence: This workflow will be configured to be executed inside a CI pipeline after the test or build workflow
```
Comment (albertdb) [inline documentation/src/github/setup-quality-pipeline.asciidoc:3]:
```suggestion
:trigger_sentence: This workflow will be configured to be executed inside a CI pipeline after the test (or build, if no test) workflow

```

File: documentation/src/github/setup-quality-pipeline.asciidoc [hunk 4]
```diff
@@ -0,0 +1,10 @@
+:provider: Github
+:pipeline_type: workflow
+:trigger_sentence: This workflow will be configured to be executed inside a CI pipeline after the test or build workflow
+:pipeline_type2: Github action
```
Comment (albertdb) [inline documentation/src/github/setup-quality-pipeline.asciidoc:4]:
```suggestion
:pipeline_type2: GitHub action

```

File: scripts/pipelines/github/templates/quality/quality-pipeline.yml.template [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  Quality:
+    name: SonarQube analysis
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+
+
+      - name: Download artifact from this workflow
```
Comment (albertdb) [inline scripts/pipelines/github/templates/quality/quality-pipeline.yml.template:30]:
```suggestion
      - name: Download artifact from build job (current workflow)
```

File: scripts/pipelines/github/templates/quality/quality-pipeline.yml.template [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/download-artifact@v2
+        id: dowloadFromSamePipeline
+        continue-on-error: true
+        with:
+          name: BuildOutput
+          path: ${{ env.targetPath }}
+
+      - name: Download artifact from build pipeline
```
Comment (albertdb) [inline scripts/pipelines/github/templates/quality/quality-pipeline.yml.template:38]:
```suggestion
      - name: Download artifact from build workflow
```

File: scripts/pipelines/github/templates/quality/quality-pipeline.yml.template [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        continue-on-error: true
+        if: ${{ steps.dowloadFromSamePipeline.outcome == 'failure' }}
+        with:
+          workflow: ${{ env.buildPipelineYamlName }}
+          commit: ${{ github.sha }}
+          check_artifacts:  true
+          name: BuildOutput
+          path: ${{ env.targetPath }}
+
+      - name: Download artifact from ci pipeline
```
Comment (albertdb) [inline scripts/pipelines/github/templates/quality/quality-pipeline.yml.template:50]:
```suggestion
      - name: Download artifact from CI workflow
```

**Review by albertdb — CHANGES_REQUESTED** [review]

File: documentation/src/azure-devops/setup-quality-pipeline.asciidoc [hunk 8]
```diff
@@ -0,0 +1,12 @@
+:provider: Azure DevOps
+
```
Comment (albertdb) [inline documentation/src/azure-devops/setup-quality-pipeline.asciidoc:2]:
```suggestion
```

File: documentation/src/github/setup-quality-pipeline.asciidoc [hunk 9]
```diff
@@ -0,0 +1,13 @@
+:provider: GitHub
+
+:pipeline_type: workflow
+:trigger_sentence: This workflow will be configured to be executed inside a CI pipeline after the test (or build, if no test) workflow
+
```
Comment (albertdb) [inline documentation/src/github/setup-quality-pipeline.asciidoc:5]:
```suggestion
```

File: documentation/src/github/setup-quality-pipeline.asciidoc [hunk 10]
```diff
@@ -0,0 +1,13 @@
+:provider: GitHub
+
+:pipeline_type: workflow
+:trigger_sentence: This workflow will be configured to be executed inside a CI pipeline after the test (or build, if no test) workflow
+
+:pipeline_type2: GitHub action
+
```
Comment (albertdb) [inline documentation/src/github/setup-quality-pipeline.asciidoc:7]:
```suggestion
```

**Review by albertdb — APPROVED** [review]

**Pull request merged**